tests/run/static_method_inheritance.pxd

```diff
@@ -0,0 +1,14 @@
+cdef class A:
```

Comment:
I think this should be in addition to the existing tests rather than replacing them.

So add a couple of new classes with `@staticmethod`s.

Otherwise it feels like we might be losing a test for some other behaviour.